src/views/payment-sheet-views/card-view.js

```diff
@@ -70,6 +79,44 @@ CardView.prototype._initialize = function () {
   }.bind(this));
 };
 
+CardView.prototype._setupCardholderName = function (cardholderNameField) {
+  var cardholderNameOptions = this.model.merchantConfiguration.card && this.model.merchantConfiguration.card.cardholderName;
+  var cardholderNameContainer = cardholderNameField.querySelector('.braintree-form__hosted-field');
+
+  this.cardholderNameInput.addEventListener('keyup', function () {
+    var hasContent = this.cardholderNameInput.value.length > 0;
+
+    classlist.toggle(cardholderNameContainer, 'braintree-form__field--valid', hasContent);
+
+    if (!cardholderNameOptions.required) {
+      return;
+    }
+
+    if (hasContent) {
+      classlist.remove(cardholderNameField, 'braintree-form__field-group--has-error');
+    }
+
+    this._sendRequestableEvent();
+  }.bind(this), false);
+
+  if (cardholderNameOptions.required) {
+    this.cardholderNameInput.addEventListener('blur', function () {
+      if (this.cardholderNameInput.value.length === 0) {
+        classlist.add(cardholderNameField, 'braintree-form__field-group--has-error');
```

Comment:
We should change this to only show the error when another field is focused like we do for Hosted Fields